src/text-editor-component.js

```diff
@@ -2884,10 +2884,25 @@ class TextEditorComponent {
 
   // Ensure the spatial index is populated with rows that are currently visible
   populateVisibleRowRange (renderedStartRow) {
-    const editorHeightInTiles = this.getScrollContainerHeight() / this.getLineHeight()
-    const visibleTileCount = Math.ceil(editorHeightInTiles) + 1
-    const lastRenderedRow = renderedStartRow + (visibleTileCount * this.getRowsPerTile())
-    this.props.model.displayLayer.populateSpatialIndexIfNeeded(Infinity, lastRenderedRow)
+    const {model} = this.props
+    const previousScreenLineCount = model.getApproximateScreenLineCount()
+
+    const renderedEndRow = renderedStartRow + (this.getVisibleTileCount() * this.getRowsPerTile())
+    this.props.model.displayLayer.populateSpatialIndexIfNeeded(Infinity, renderedEndRow)
+
+    // It is possible for the approximate screen line count to exceed the actual
```

Comment:
The first sentence of this comment is confusing to me. Isn't the problem that the approximate screen line count is actually *less* than the actual screen line count because it's not accounting for all the soft wraps? I wonder if we should just always clear the dimensions cache when going down this code path for simplicity.